default.nix

```diff
@@ -9,106 +9,27 @@
 { system ? builtins.currentSystem
 , crossSystem ? null
 , config ? { allowUnfreePredicate = (import ./lib.nix).unfreePredicate; }
-
   # Overrides for niv
 , sourcesOverride ? { }
   # Nixpkgs override
-, pkgs ? import ./nix/default.nix { inherit system crossSystem config sourcesOverride; }
-
+, pkgs ? import ./nix { inherit system crossSystem config sourcesOverride; }
+  # local packages
+, _pkgs ? import ./nix/pkgs { inherit rev pkgs checkMaterialization; }
```

Comment:
Maybe `localPkgs`? `_pkgs` is a bit obscure.

Reply:
Also the split here isn't totally clear to me: some things are in `pkgs` provided by overlays, but only consume `pkgs`. So why not put them in `_pkgs`? It's a bit mysterious.

Reply:
pkgs = pinned nixpkgs with various overlays consumed by the project
pkgsLocal = local packges used for development, ie none of those are being deployed anywhere

Reply:
Fine: but we don't need to stuff all external things into overlays. Overlays are a particularly obscure way to make an attribute set by putting more things into nixpkgs. IMO, if you're not overriding something from nixpkgs with your overlay, it probably shouldn't be an overlay.

I guess it seems to me that we have three classes of things:
- Really external stuff: we just pull in the definition. e.g. nixpkgs, niv stuff
- "Morally" external stuff but where we define it locally because it doesn't exist externally. e.g. the extra sphinx packages
- Stuff where the software itself is defined in this repo. e.g. the Haskell packages

Bundling the first two together seems reasonable, though.

Reply:
We discussed this on Slack. I'll add appropriate comments on how this PR/ this work is going to be continued